src/util/SpawnPed.cpp

```diff
@@ -0,0 +1,50 @@
+#include "core/frontend/Notifications.hpp"
+#include "game/backend/FiberPool.hpp"
+#include "game/backend/ScriptMgr.hpp"
+#include "game/rdr/Natives.hpp"
+
+
+namespace YimMenu
+{
+	bool SpawnPed(std::string model_name, Player player, int playerPed)
+	{
+		FiberPool::Push([=] {
+			Hash model = MISC::GET_HASH_KEY(model_name.c_str());
+			LOG(INFO) << "FOUND HASH";
+			if (STREAMING::IS_MODEL_IN_CDIMAGE(model) && STREAMING::IS_MODEL_VALID(model))
+			{
+				if (!STREAMING::HAS_MODEL_LOADED(model))
```

Comment:
Make this either a for loop or while loop to allow a longer window for the game to stream in the models. Integrate a timeout in case it breaks somehow. Coupled with `IS_MODEL_VALID` this should just work fine, but you never know.

Reply:
a while loop and for loop do not work in a better manner here. I have tested it and the if statement works better.